programs/clearing_house/src/math/position.rs

```diff
@@ -84,10 +84,33 @@ pub fn calculate_base_asset_value(
         .checked_sub(amm.user_lp_shares)
         .ok_or_else(math_error!())?;
 
-    let base_asset_reserve_proportion =
-        get_proportion_u128(base_asset_reserve, amm_lp_shares, amm.sqrt_k)?;
-    let quote_asset_reserve_proportion =
-        get_proportion_u128(quote_asset_reserve, amm_lp_shares, amm.sqrt_k)?;
+    let base_asset_reserve_proportion = if amm.user_lp_shares == 0 {
```

Comment:
does this need to be patched to master? if its broken on master should fix there sooner than later

Reply:
can you add tests for the functions, there are a lot of forking in here

Reply:
reverted this since made opt within `get_proportion_u128`